x-pack/plugins/ml/public/jobs/jobs_list/components/job_filter_bar/job_filter_bar.js

```diff
@@ -41,14 +45,37 @@ export class JobFilterBar extends Component {
   constructor(props) {
     super(props);
 
+    this.state = { error: null };
     this.setFilters = props.setFilters;
   }
 
-  onChange = ({ query }) => {
-    const clauses = query.ast.clauses;
-    this.setFilters(clauses);
+  onChange = ({ query, error }) => {
+    if (error) {
+      this.setState({ error });
+    } else {
+      const clauses = query.ast.clauses;
```

Comment:
for additional safety, it might be best to wrap this in a check that `query.ast` and `query.ast.clauses` are not `undefined`.
it wouldn't need to display an error, just show no results.